src/librustc/ty/sty.rs

```diff
@@ -1848,6 +1848,73 @@ impl<'a, 'gcx, 'tcx> TyS<'tcx> {
             _ => bug!("cannot convert type `{:?}` to a closure kind", self),
         }
     }
+
+    //
+    pub fn has_specified_layout(&self, tcx: TyCtxt<'a, 'gcx, 'tcx>) -> Option<bool> {
+        match self.sty {
+            // Not concrete, cannot say anything
+            TyInfer(_) | TyError | TyParam(_) | TyProjection(_) | TyAnon(_, _) | TyDynamic(_, _) =>
+                None,
+            // Outright unspecified to be anything in particular
+            TyClosure(_, _) | TyGenerator(_, _, _) | TyGeneratorWitness(_) => Some(false),
+            TyFnDef(_, _) => Some(false),
```

Comment:
Aren't function definitions guaranteed to be zst? (Not that it's useful to transmute them... but still)

Reply:
I don't recall us specifying that at any point (and in fact, I remember us debating whether we should make fndef types be a pointer in the past).

Either way, I’m willing to filter out those cases, if they occur on a crater run, manually.

Reply:
We had a warning period when we made them ZSTs, see also `E0591` in `intrinsicck`.
I think `Some(false)` here is decent.